src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/FrozenDictionary.cs

```diff
@@ -268,12 +273,12 @@ public static class FrozenDictionary
 
             if (source.Count <= Constants.MaxItemsInSmallFrozenCollection)
             {
-                // use the specialized dictionary for low item counts
-                return new SmallFrozenDictionary<TKey, TValue>(source, comparer);
+                // Use the specialized dictionary for low item counts.
+                return new SmallFrozenDictionary<TKey, TValue>(source);
```

Comment:
Do we no longer need to pass the comparer here?

Reply:
The information is part of the source via source.Comparer.  I removed all of the cases where there was such duplication in order to avoid accidentally flowing the wrong thing.